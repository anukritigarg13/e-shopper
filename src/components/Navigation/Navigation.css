.navigation{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    background-color: rgb(189, 159, 159);
    color: whitesmoke;
    padding: 1%;
}