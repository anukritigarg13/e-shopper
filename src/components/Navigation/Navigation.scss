@import '../styles/index.scss';

.navigation{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
    background-color:$primary-background-color;
    padding: 1% 3%;
    @include custom-mixin;
}
.navigation--dark-mode{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
    background-color: whitesmoke;
    color: rgb(75, 72, 72);
    padding: 1% 3%;
}
.navigation-links--dark-mode{
    color: rgb(75, 72, 72);
    margin: 0.5rem;
    text-decoration: none;
}
.navigation-links{  
    color: whitesmoke;
    margin: 0.5rem;
    text-decoration: none;
}
.order-nav{
    display: flex;

}
@media screen and (max-width:912px){
    .navigation{
        padding: 3% 5%;
    }
}
