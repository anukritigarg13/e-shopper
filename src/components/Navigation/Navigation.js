import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import './Navigation.scss';
import { Link } from 'react-router-dom';
import NavContext from '../../theme';

const Navigation = ({ cartItemsCount, toggleTheme }) => {
  const currentTheme = useContext(NavContext);
  return (
    <div className={currentTheme === 'light' ? 'navigation' : 'navigation--dark-mode'} data-testid="navigation">
      <div><Link to="/" className={currentTheme === 'light' ? 'navigation-links' : 'navigation-links--dark-mode'}>E-Shopper</Link></div>
      <div>
        <button type="button" onClick={toggleTheme}> Change Theme</button>
      </div>
      <div className="order-nav">
        <div><Link to="/allOrders" className={currentTheme === 'light' ? 'navigation-links' : 'navigation-links--dark-mode'}>All Orders</Link></div>
        <div>
          <Link to="/cart" className={currentTheme === 'light' ? 'navigation-links' : 'navigation-links--dark-mode'}>

            My Basket :
            {cartItemsCount}

          </Link>

        </div>
      </div>

    </div>
  );
};

Navigation.propTypes = {
  cartItemsCount: PropTypes.number.isRequired,
  toggleTheme: PropTypes.func.isRequired,
};

export default Navigation;
