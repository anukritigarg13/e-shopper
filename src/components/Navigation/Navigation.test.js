/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navigation from './Navigation';

describe('Navigation component', () => {
  afterEach(() => jest.clearAllMocks);
  const mockProps = {
    cartItemsCount: 10,
    toggleTheme: jest.fn(),
  };
  it('should display Change Theme button', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const themeToggleElement = screen.getByText('Change Theme');
    expect(themeToggleElement.tagName).toBe('BUTTON');
  });
  it('should display E-Shopper nav link', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const eShopperElement = screen.getByText('E-Shopper');
    expect(eShopperElement.tagName).toBe('A');
  });
  it('should display All Orders nav link', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const eShopperElement = screen.getByText('All Orders');
    expect(eShopperElement.tagName).toBe('A');
  });
  it('should display My Basket nav link and show count of cart items as 10', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const eShopperElement = screen.getByText('My Basket :10');
    expect(eShopperElement.tagName).toBe('A');
  });
  it('should go to /cart when My Basket clicked', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const eShopperElement = screen.getByText('My Basket :10');
    fireEvent.click(eShopperElement);
    expect(document.location.href).toBe('http://localhost/cart');
  });
  it('should go to / when E-Shopper clicked', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const eShopperElement = screen.getByText('E-Shopper');
    fireEvent.click(eShopperElement);
    expect(document.location.href).toBe('http://localhost/');
  });
  it('should go to /allOrders when All Orders clicked', () => {
    render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    const eShopperElement = screen.getByText('All Orders');
    fireEvent.click(eShopperElement);
    expect(document.location.href).toBe('http://localhost/allOrders');
  });
  it('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Navigation {...mockProps} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
});
