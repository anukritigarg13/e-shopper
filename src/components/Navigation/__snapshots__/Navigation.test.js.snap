// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match snapshot 1`] = `
<div>
  <div
    class="navigation"
    data-testid="navigation"
  >
    <div>
      <a
        class="navigation-links"
        href="/"
      >
        E-Shopper
      </a>
    </div>
    <div>
      <button
        type="button"
      >
         Change Theme
      </button>
    </div>
    <div
      class="order-nav"
    >
      <div>
        <a
          class="navigation-links"
          href="/allOrders"
        >
          All Orders
        </a>
      </div>
      <div>
        <a
          class="navigation-links"
          href="/cart"
        >
          My Basket :
          10
        </a>
      </div>
    </div>
  </div>
</div>
`;
